### PR TITLE
Update Vault export Name

### LINKED ISSUE #181: [BUG] Keygen discovery / keysign discovery screen , main device should always be selected by default

### LINKED ISSUE #204: Export Vault Name Detailed
# Context

When user create vault, not much data in name

use

```
voltix-{name}-{YYYY-MM}-{n}of{m}-{kkkk}.dat
```

YYYY: year
MM: date (number)
{n}of{m}: 2of3
kkkk: last 4 characters of pub key

image

image

### PR DESCRIPTION
**Pull request by johnnyluo**
This PR fix a few issues:
1. Update app icon to remove the alpha channel
2. Update vault export name , fix #204 
3. When keygen/keysign , automatically select the main device fix #181 
4. When keygen , display vault information  2 of 3 , 3 of 4 etc

### CODE REVIEW

File: voltixApp/VoltixApp/View Models/KeygenPeerDiscoveryViewModel.swift [hunk 1]
```diff
@@ -27,6 +27,7 @@ class KeygenPeerDiscoveryViewModel: ObservableObject {
     @Published var localPartyID = ""
     @Published var selections = Set<String>()
     @Published var serverAddr = "http://127.0.0.1:8080"
+    @Published var vaultDetail = ""
```
Comment (enriquesouza) [inline voltixApp/VoltixApp/View Models/KeygenPeerDiscoveryViewModel.swift:30]: can we use that enum .empty?

Comment (enriquesouza) [reply voltixApp/VoltixApp/View Models/KeygenPeerDiscoveryViewModel.swift:30]: @Published var vaultDetail = .empty

**Review by enriquesouza — APPROVED** [review]

File: voltixApp/VoltixApp/Views/Keygen/PeerDiscoveryView.swift [hunk 2]
```diff
@@ -194,12 +198,24 @@ struct PeerDiscoveryView: View {
...
-            viewModel.selections.remove(peer)
+            if peer != viewModel.localPartyID {
+                viewModel.selections.remove(peer)
+            }
         } else {
             viewModel.selections.insert(peer)
         }
+        let totalSigners = viewModel.selections.count
+        if totalSigners >= 3 {
+            let threshold = Int(ceil(Double(totalSigners) * 2.0 / 3.0))
```
Comment (flypaper0) [inline voltixApp/VoltixApp/Views/Keygen/PeerDiscoveryView.swift:216]: what does it do?

File: voltixApp/VoltixApp/Views/Keysign/KeysignDiscoveryView.swift [hunk 3]
```diff
@@ -196,12 +196,15 @@ struct KeysignDiscoveryView: View {
         return Image(systemName: "xmark")
     }
     
-    private func handleSelection(_ peer: String) {
+    func handleSelection(_ peer: String) {
         isLoading = true
         
         DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
```
Comment (flypaper0) [inline voltixApp/VoltixApp/Views/Keysign/KeysignDiscoveryView.swift:202]: why we need asyncAfter?

**Review by flypaper0 — APPROVED** [review]

**Review by Rockindash — APPROVED** [review]

File: voltixApp/VoltixApp/Views/Keygen/PeerDiscoveryView.swift (hunk 2 above)
Comment (johnnyluo) [reply voltixApp/VoltixApp/Views/Keygen/PeerDiscoveryView.swift:216]: it calculate the threshold , number of signers to do a keysign

File: voltixApp/VoltixApp/Views/Keysign/KeysignDiscoveryView.swift (hunk 3 above)
Comment (johnnyluo) [reply voltixApp/VoltixApp/Views/Keysign/KeysignDiscoveryView.swift:202]: Not exactly sure about that , @Rockindash probably knows

**Pull request merged**